addons/product/migrations/9.0.1.2/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,183 @@
+---Fields in module 'product'---
+product      / product.attribute        / attribute_line_ids (one2many) : NEW relation: product.attribute.line
+# New link to counterpart of existing many2many relation
+product      / product.attribute        / sequence (integer)            : NEW 
+# New field ---> Nothing to do
+
+product      / product.packaging        / code (char)                   : DEL
+product      / product.packaging        / ean (char)                    : DEL 
+product      / product.packaging        / name (text)                   : now required
+product      / product.packaging        / name (text)                   : type is now 'char' ('text')
+product      / product.packaging        / rows (integer)                : DEL required: required, req_default: 3
+product      / product.packaging        / ul (many2one)                 : DEL relation: product.ul, required: required, req_default: function
+product      / product.packaging        / ul_container (many2one)       : DEL relation: product.ul
+# TODO: remove NOT NULL constraint on these obsolete fields
```

Comment:
OpenUpgrade, unlike Odoo does not drop obsolete columns at the end of a module upgrade. That means we have to remove these constraints to prevent problems when a new product is added in the migrated production environment (before using something like database_cleanup). You are right, I don't remember this coming up before.

We can probably do this globally in the exact same place where Odoo drops the column instead of on a per module basis.